extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -584,54 +637,89 @@ export class AmpAnalytics extends AMP.BaseElement {
 
     const resultPromises = [];
     for (let r = 0; r < requests.length; r++) {
-      const request = this.requests_[requests[r]];
-      resultPromises.push(this.handleRequestForEvent_(request, trigger, event));
+      const requestName = requests[r];
+      resultPromises.push(
+          this.handleRequestForEvent_(requestName, trigger, event));
     }
     return Promise.all(resultPromises);
   }
 
   /**
+   * Check request is v1 or v2
+   * @param {string} request
+   * @return {boolean}
+   */
+  isV2Request_(request) {
+    if (!this.config_['requests'] ||
+        !hasOwn(this.config_['requests'], request)) {
+      if (this.config_['requests-v2'] &&
+          hasOwn(this.config_['requests-v2'], request)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
    * Processes a request for an event callback and sends it out.
    *
-   * @param {string} request The request to process.
+   * @param {string} requestName The requestName to process.
    * @param {!JsonObject} trigger JSON config block that resulted in this event.
    * @param {!Object} event Object with details about the event.
    * @return {!Promise<string|undefined>} The request that was sent out.
    * @private
    */
-  handleRequestForEvent_(request, trigger, event) {
+  handleRequestForEvent_(requestName, trigger, event) {
     if (!this.element.ownerDocument.defaultView) {
       const TAG = this.getName_();
       dev().warn(TAG, 'request against destroyed embed: ', trigger['on']);
       return Promise.resolve();
     }
 
+    let request = null;
+    let isV2 = false;
+    if (this.isV2Request_(requestName)) {
+      request = this.requestsV2_[requestName];
+      isV2 = true;
+    } else {
+      request = this.requests_[requestName];
+    }
+
     if (!request) {
       const TAG = this.getName_();
       this.user().error(TAG, 'Ignoring event. Request string ' +
           'not found: ', trigger['request']);
       return Promise.resolve();
     }
 
-    return this.checkTriggerEnabled_(trigger, event)
-        .then(enabled => {
-          if (!enabled) {
-            return;
-          }
-          return this.expandAndSendRequest_(request, trigger, event);
-        });
+    return this.checkTriggerEnabled_(trigger, event).then(enabled => {
+      if (!enabled) {
+        return;
+      }
+      return this.expandAndSendRequest_(request, trigger, event, isV2);
+    });
   }
 
   /**
-   * @param {string} request The request to process.
+   * @param {string|RequestHandler} request The request to process.
    * @param {!JsonObject} trigger JSON config block that resulted in this event.
    * @param {!Object} event Object with details about the event.
-   * @return {!Promise<string>} The request that was sent out.
+   * @param {boolean=} opt_v2 Is request v2.
```

Comment:
For example you wouldn't have to pass around a boolean like this, you can just check the request object for the same information.